### ISSUE TITLE
Blurred blue lines in series mode

### ISSUE THREAD

**Issue by zb-markus** [klokantech/kartenportal-tickets#215]
The blue lines in the series mode are blurred, sometimes only in a certain zoom level. Some lines are also darker than others. When you zoom in as much as possible, you always see a single line as expected. It is therefore a rendering problem, not one of the bounding boxes within the metadata. A possible solution could be to draw the lines always in an even amount of pixels, e.g. 4 or 6 pixels. 

An image for upload follows, made at this position:
http://klokan.github.io/maprank-client/#series&id=195467248 

Bug seen in IE8.

**Comment by klokan** [comment]
If there are multiple editions of the same map they are all rendered over each other. This makes some of the sheets fill more white and blue lines more visible. If you click on the sheet you will see the editions listed in the detail window now.

If you provide is with the screenshot I can tell if it is expected behaviour or not.

**Comment by klokan** [comment]
![blurred_lines](https://f.cloud.github.com/assets/59284/1742944/fa2078c4-6404-11e3-8535-de135d2f889c.gif)

**Comment by klokan** [comment]
This is quite expected. Sheet Q-37 has 3 editions while the sheets around have just a single editions, so Q-37 is slightly more white and with darker border.

Note: I can try to experiment with quality of antialiasing (blurriness) of lines by changing the settings of 'gamma' of the render and removing transparency on lines.

**Comment by zb-markus** [comment]
A "normal" user cannot understand this and thinks that the rendering of the blue lines is not state of the art. It does not look "good". We expect that all lines are:
- even in thickness
- evenly sharp, not blurred
- even in colour
- even in transparency

Note: a separate request "Map series and time filter synchronised" moved to a new ticket #218

**Comment by klokan** [comment]
On your screenshot of are artefacts not visible on our screen. Are you able to see the problems on other computers as well or is it related to the Windows terminals in ZB only?

This is how the same screenshot looks under the latest Chrome:

![screen shot 2013-12-13 at 6 19 51 pm](https://f.cloud.github.com/assets/59284/1744406/885d30de-641c-11e3-82dd-955c2202c381.png)

and how it looks in the Windows XP in IE7:

![screen shot 2013-12-13 at 18 30 39](https://f.cloud.github.com/assets/59284/1747170/68db1ad8-6452-11e3-8ea4-3fce4fb438ae.png)

We are using a server side drawing of GIS data - from a database with close to 300.000 records and series with up to 9300 sheets - and it is a state of the art quality of rendering without doubts. The same rendering engine is in use by OpenStreetMap, MapQuest, MapBox, TileMill, etc.

I have been now for 3 hours testing the alternative configurations of the renderer and slightly improved the quality.

The bounding boxes for sheets are available in geographical coordinates and as such there is rounding of float numbers and antialiasing involved during rendering on the screen. The advantage is that we can also directly render general polygons or shapes, not only rectangles. Part of the map indexes are the visible sheet numbers as well - with the intelligent placement and user interaction.

All the sheets from the metadata database are displayed and the system provides functionality exactly according the specification from the contract for ETH Zurich - based on the specification we estimated required time and budget. The available time is now heading to the end. We have delivered complete time sheets so it is very transparent what was done any how much effort it took us to build the system available now.

The system can be of course improved further via new enhancements if you order these.

The mentioned conditions for:
- even in thickness
- evenly sharp, not blurred
- even in colour
- even in transparency

were not discussed until now (and the work is more or less finished at this point), were not expected from our side for the implementation, were not specified in the contract, and were not estimated in the budget.

In fact if we should reach the mentioned target there is substantially more work involved because:
- duplicate sheets must be recognised during preprocessing on the server side and only a single candidate sheet is selected to be drawn in the map index to represent a position. Now we draw all of the sheets in the series.
- the edges inside of the grid where sheets are "touching" are drawn twice - because sheets are using geometrical representation on a computer (model a la ESRI ShapeFile). If the edges should be shared between the two sheets and drawn only once then we would need to recalculate everything into topological representation and render it as such.
- now we are using server side rendering of the sheets. This means even IE7 or IE8 and old computers runs very well even on large map series (over 9000 sheets). If we would choose the client side rendering the edges could be possibly sharper, but we would have to get rid of the support of old Internet Explorers and very old computers, because these are technically not able to draw 9000 shapes as a map overlay.

All of the suggestions are great ideas and I like these - and we can discuss possible implementation in future, but at this point I definitely don't consider the available look&feel to be a bug.

If there is time remaining in the contract after all bug reports are fixed we can work on enhancements as well.

**Comment by ethbib-susanne** [comment]
J agree that we should work on this problem if there is time remaining.

**Comment by klokan** [comment]
Even transparency implemented for #227.
